core/src/main/java/org/mule/registry/DefaultRegistryBroker.java

```diff
@@ -7,41 +7,75 @@
 package org.mule.registry;
 
 import org.mule.api.MuleContext;
+import org.mule.api.registry.LifecycleRegistry;
 import org.mule.api.registry.Registry;
 
 import java.util.Collection;
 import java.util.List;
 import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicReference;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @deprecated as of 3.7.0. This will be removed in Mule 4.0
+ */
+@Deprecated
 public class DefaultRegistryBroker extends AbstractRegistryBroker
 {
-    private TransientRegistry transientRegistry;
-    private List<Registry> registries = new CopyOnWriteArrayList<Registry>();
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultRegistryBroker.class);
+
+    private final List<Registry> registries = new CopyOnWriteArrayList<>();
+    private final AtomicReference<LifecycleRegistry> lifecycleRegistry = new AtomicReference<>(null);
 
     public DefaultRegistryBroker(MuleContext context)
     {
         super(context);
-        transientRegistry = new TransientRegistry(context);
-        registries.add(0, transientRegistry);
-    }
-
-    TransientRegistry getTransientRegistry()
-    {
-        return transientRegistry;
+        addRegistry(new SimpleRegistry(context));
     }
 
+    @Override
     public void addRegistry(Registry registry)
     {
-        registries.add(1, registry);
+        registries.add(0, registry);
+        if (registries.size() > 2 && LOGGER.isWarnEnabled())
+        {
+            LOGGER.warn(registries.size() + " registries have been registered, however adding registries has been deprecated as of Mule 3.7.0");
+        }
     }
 
+    @Override
     public void removeRegistry(Registry registry)
     {
         registries.remove(registry);
+        if (registry instanceof LifecycleRegistry)
+        {
+            lifecycleRegistry.compareAndSet((LifecycleRegistry) registry, null);
+        }
     }
 
-    protected Collection<Registry> getRegistries()
+    @Override
+    public Collection<Registry> getRegistries()
     {
         return registries;
```

Comment:
Now that this is public, it would be better to return an immutable copy of the registries instance